cmd/timestamp-cli/app/verify.go

```diff
@@ -67,27 +75,143 @@ func runVerify() (interface{}, error) {
 		return nil, fmt.Errorf("error reading request from file: %w", err)
 	}
 
-	certChainPEM := viper.GetString("cert-chain")
-	pemBytes, err := os.ReadFile(filepath.Clean(certChainPEM))
+	artifactPath := viper.GetString("artifact")
+	artifact, err := os.Open(filepath.Clean(artifactPath))
 	if err != nil {
-		return nil, fmt.Errorf("error reading request from file: %w", err)
+		return nil, err
+	}
+
+	opts, err := newVerifyOpts()
+	if err != nil {
+		return verifyCmdOutput{TimestampPath: tsrPath}, fmt.Errorf("failed to created VerifyOpts: %w", err)
+	}
+
+	err = verification.VerifyTimestampResponse(tsrBytes, artifact, opts)
+
+	return &verifyCmdOutput{TimestampPath: tsrPath}, err
+}
+
+func newVerifyOpts() (verification.VerifyOpts, error) {
+	opts := verification.VerifyOpts{}
+
+	oid, err := getOID()
+	if err != nil {
+		return verification.VerifyOpts{}, fmt.Errorf("failed to parse value from oid flag: %w", err)
 	}
+	if oid != nil {
+		opts.OID = oid
+	}
+
+	certPathFlagVal := viper.GetString("certificate")
+	if certPathFlagVal != "" {
+		cert, err := parseTSACertificate(certPathFlagVal)
+		if err != nil {
+			return verification.VerifyOpts{}, fmt.Errorf("failed to parse cert flag value from PEM file: %w", err)
+		}
+		opts.TSACertificate = cert
+	}
+
+	roots, intermediates, err := getRootAndIntermediateCerts()
+	if err != nil {
+		return verification.VerifyOpts{}, fmt.Errorf("failed to parse root and intermediate certs from certificate-chain flag: %w", err)
+	}
+	if roots != nil && intermediates != nil {
```

Comment:
It should be fine to unconditionally set the options. Also if there's only a root provided and no intermediates, will this conditional be skipped?

Reply:
Yes it will be skipped, I'll update that if statement.